### PR TITLE
docs: add secure boot verification

### PR DESCRIPTION
**Pull request by m-iwanicki**
Rebase and change target to main before merge

### CODE REVIEW

**Comment by pietrushnic** [comment]
@macpijan I wonder what you think about its design. It closely relates to my teaching materials for upcoming hardwear.io and OST2 training.

**Review by pietrushnic — CHANGES_REQUESTED** [review]

File: docs/guides/secure-boot/secure-boot-verification.md [hunk 1]
```diff
@@ -0,0 +1,671 @@
...
+
+* Restore BIOS to defaults.
+    <!--
+    Does it also restore SB keys?
+    if not then SBO006.001 + restore should probably be first test so we start
+    from known state
+    -->
+* Attached USB drive prepared in [USB drive](#usb-drive) section
+    - If using QEMU `tests.img` can be attached directly e.g. via adding
+        `-drive if=ide,file=tests.img` argument to `qemu` command
```
[the remark this answers is not in the document]
Comment (m-iwanicki) [reply docs/guides/secure-boot/secure-boot-verification.md:13]: Script creates FAT image that contains EFI partition and uses generated UUID later to find correct USB drive and add all `.efi` files to boot options.
If it's mounted as directory then `add-boot-options.sh` would need some changes.

File: docs/guides/secure-boot/secure-boot-verification.md [hunk 2]
```diff
@@ -0,0 +1,671 @@
...
+```
+
+<!--
+    Another way:
+
+    ```shell
+    mokutil --sb-state
+    SecureBoot enabled
+    ```
+-->
```
[the remark this answers is not in the document]
Comment (pietrushnic) [reply docs/guides/secure-boot/secure-boot-verification.md:210]: It looks like that state comes from UEFI Variables. I guess the OS cannot do anything better; maybe expect attesting firmware to present that state.

**Comment by macpijan** [comment]
> @macpijan I wonder what you think about its design. It closely relates to my teaching materials for upcoming hardwear.io and OST2 training.

I think my main concert was duplication with content from docs.dasharo.com

**Comment by pietrushnic** [comment]
@macpijan, what would be our preferred way to address the duplication issue? I have to say that the description docs.dasharo.com is outdated and should be corrected. Also, the methodology should be changed not to use UEFI Secure Boot, as done here.

I agree we need some consistency, and maybe this documentation better fits Dasharo as it tests its implementation of OTOH; it should also test OVMF and any proprietary implementation. Finally, I would like to make this implementation a base for future products to assess the quality of UEFI Secure Boot implementation. In that light, what would be the best way to deal with this PR?

**Comment by artur-rs** [comment]
Merging the documents in this form, as the Dasharo docs sync and test-case update will be prepared in the next phase of this project.

**Pull request merged**